jax/core.py

```diff
@@ -161,17 +187,22 @@ class JaxprEqn(NamedTuple):
   outvars: List['Var']
   primitive: 'Primitive'
   params: Dict[str, Any]
+  effects: Effects
   source_info: source_info_util.SourceInfo
 
   def __repr__(self):
     return str(pp_eqn(self, JaxprPpContext(), custom_pp_eqn_rules=False)
                ).rstrip()
 
-def new_jaxpr_eqn(invars, outvars, primitive, params, source_info=None):
+  def replace(self, *args, **kwargs):
+    return self._replace(*args, **kwargs)
+
+def new_jaxpr_eqn(invars, outvars, primitive, params, effects, source_info=None):
+  effects = effects or set()
```

Comment:
Do we want to support any falsey value to be passed to new_jaxpr_eqn? Maybe better not to add that polymorphism and require the caller to pass `set()`.

Reply:
Done.